pkg/policy/directory/watcher.go

```diff
@@ -0,0 +1,217 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return cnp, nil
+}
+
+func getLabels(fileName string, cnp *cilium_v2.CiliumNetworkPolicy) labels.LabelArray {
+	labelsArr := labels.LabelArray{
+		labels.NewLabel("filename", fileName, labels.LabelSourceDirectory),
+	}
+
+	ns := cnp.ObjectMeta.Namespace
+	// For clusterwide policy namespace will be empty.
+	if ns != "" {
+		nsLabel := labels.NewLabel(k8sConst.PolicyLabelNamespace, ns, labels.LabelSourceDirectory)
+		labelsArr = append(labelsArr, nsLabel)
+		srcLabel := labels.NewLabel("policy-derived-from", k8sCiliumUtils.ResourceTypeCiliumNetworkPolicy, labels.LabelSourceDirectory)
+		labelsArr = append(labelsArr, srcLabel)
+	} else {
+		srcLabel := labels.NewLabel("policy-derived-from", k8sCiliumUtils.ResourceTypeCiliumClusterwideNetworkPolicy, labels.LabelSourceDirectory)
+		labelsArr = append(labelsArr, srcLabel)
+	}
+
+	return labelsArr
+}
+
+// read cilium network policy yaml file and convert to policy object and
+// add rules to policy engine.
+func (p *policyWatcher) addToPolicyEngine(cnpFilePath string) error {
+	// read from file and convert to cnp object
+	cnp, err := p.translateToCNPObject(cnpFilePath)
+	if err != nil {
+		return err
+	}
+
+	cnp.ObjectMeta.Name = filepath.Base(cnpFilePath)
+
+	resourceID := ipcacheTypes.NewResourceID(
+		ipcacheTypes.ResourceKindFile,
+		cnp.ObjectMeta.Namespace,
+		cnp.ObjectMeta.Name,
+	)
+
+	// convert to rules
+	rules, err := cnp.Parse()
+	if err != nil {
+		return err
+	}
+
+	//update labels
+	lbls := getLabels(cnp.ObjectMeta.Name, cnp)
+	for _, r := range rules {
+		r.Labels = lbls
+	}
+
+	// add to policy engine
+	_, err = p.policyManager.PolicyAdd(rules, &policy.AddOptions{
+		ReplaceByResource: true,
+		Source:            source.Directory,
+		Resource:          resourceID,
+	})
+
+	if err == nil {
+		p.fileNameToCnpCache[cnp.ObjectMeta.Name] = cnp
+	}
+
+	return err
+}
+
+func (p *policyWatcher) deleteFromPolicyEngine(cnpFilePath string) error {
+	fileName := filepath.Base(cnpFilePath)
+	cnp := p.fileNameToCnpCache[fileName]
+	if cnp == nil {
+		p.log.WithField("file", fileName).Warn("map entry doesn't exist")
+		return nil
+	}
+
+	resourceID := ipcacheTypes.NewResourceID(
+		ipcacheTypes.ResourceKindFile,
+		cnp.ObjectMeta.Namespace,
+		fileName,
+	)
+	_, err := p.policyManager.PolicyDelete(getLabels(fileName, cnp), &policy.DeleteOptions{
+		Source:   source.Directory,
+		Resource: resourceID})
+
+	delete(p.fileNameToCnpCache, fileName)
+	return err
+}
+
+func (p *policyWatcher) watchDirectory(ctx context.Context) {
+	go func() {
+		p.log.Info("Directory policy watcher started")
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			p.log.WithField("Err", err).Error("Initializing NewWatcher failed")
+			return
+		}
+		defer watcher.Close()
+
+		dir := p.config.StaticCiliumNetworkPolicyPath
+		err = watcher.Add(dir)
+		if err != nil {
+			p.log.WithFields(logrus.Fields{"Dir": dir, "err": err}).Error("Directory Watcher Add failed")
```

Comment:
nit: 

We're using `"Dir"` here but `"dir"` below, that's an inconsistency . I'd recommend using `logfields.Path` as a key instead. Also please use `WithError(err)`:

```suggestion
			p.log.WithError(err).WithField(logfields.Path, dir).Error("Failed to watch policy directory. Policies will not be loaded from disk")
```